app/models/concerns/media_apify_item.rb

```diff
@@ -5,42 +5,82 @@ class ApifyError < StandardError; end
   class ApifyResponseError < StandardError; end
 
   Media.class_eval do
-    def self.apify_start_request(url)
-      apify_url = "https://api.apify.com/v2/acts/apify~facebook-posts-scraper/run-sync-get-dataset-items?token=#{PenderConfig.get('apify_api_token')}"
+    APIFY_BASE_URL = "https://api.apify.com/v2/acts"
+    APIFY_SUFFIX_URL = "/run-sync-get-dataset-items?token=#{PenderConfig.get('apify_api_token')}"
 
-      headers = {
-        'content-type' => 'application/json'
-      }
+    SCRAPERS = {
+      facebook: "apify~facebook-posts-scraper",
+      instagram: "apify~instagram-scraper"
+    }.freeze
 
-      payload = {
-        resultsLimit: 20,
-        startUrls: [
-          { url: url }
-        ]
-      } 
+    def self.apify_start_request(url, platform = :facebook)
+      apify_url = build_apify_url(platform)
+      raise ApifyError, 'Unsupported platform' unless apify_url
 
+      payload = build_payload(url, platform)
+      response = make_apify_request(apify_url, payload)
+
+      handle_response(response, apify_url)
+    end
+
+    def self.build_apify_url(platform)
+      scraper = SCRAPERS[platform]
+      raise ApifyError, 'Unsupported platform' unless scraper
+
+      "#{APIFY_BASE_URL}/#{scraper}#{APIFY_SUFFIX_URL}"
+    end
+
+    def self.build_payload(url, platform)
+      case platform
+      when :facebook
+        {
+          resultsLimit: 20,
+          startUrls: [
+            { url: url }
+          ]
+        }
+      when :instagram
+        {
+          addParentData: false,
+          directUrls: [url],
+          enhanceUserSearchWithFacebookPage: false,
+          isUserReelFeedURL: false,
+          isUserTaggedFeedURL: false,
```

Comment:
Question: Can you reliably assume this?

Reply:
Good question. I've removed these checks since we cannot assume this.